### ISSUE TITLE
Conda environment + setup

### ISSUE THREAD

**Issue by richardkoehler** [neuromodulation/py_neuromodulation#144]
@timonmerk 
I tried to install pyneuromodulation as package, but it wasn't so easy to make it work. I did the following:

- I modified env.yml (previously settting up the environment didn't work - and additionally env.yml was configured to install all packages via pip, now they are (except pybids) installed via conda.
- I installed the conda environment via 
> conda env create -f env.yml 
- I activated the conda environment via
> conda activate pyneuromodulation_test 
- I installed pyneuromodulation in editable mode
> pip install -r requirements.txt --editable . 
- In python, I ran
> import py_neuromodulation
> print(dir(py_neuromodulation)) 
- Which told me that the only items in this package are
> ['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__'] <
- Only once I had added the line: "from . import nm_BidsStream" to __init__.py, did I get the following output for dir():
>['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'nm_BidsStream', 'nm_IO', 'nm_bandpower', 'nm_coherence', 'nm_define_nmchannels', 'nm_eval_timing', 'nm_features', 'nm_fft', 'nm_filter', 'nm_generator', 'nm_hjorth_raw', 'nm_kalmanfilter', 'nm_normalization', 'nm_notch_filter', 'nm_plots', 'nm_projection', 'nm_rereference', 'nm_resample', 'nm_run_analysis', 'nm_sharpwaves', 'nm_stft', 'nm_stream', 'nm_test_settings']

Questions:
- Was this the intended way of setting up the environment and the package?
- What could the problem with __init__.py be related to?

**Comment by richardkoehler** [comment]
I created a branch to work on this here:
https://github.com/neuromodulation/py_neuromodulation/tree/setup-package

**Comment by timonmerk** [comment]
Thanks a lot @richardkoehler for notifying on this issue. I adapted the ReadMe.rst and deleted the conda environment setup "--user" flag. For some reason this option yielded an error.

Additionally i added some packages that were missing in the requirements.txt. 

For the print(dir(py_neuromodulation)) I got the same result. But following the example_BIDS.ipynb you should still be able to import the files nevertheless: 

<img width="1380" alt="image" src="https://user-images.githubusercontent.com/38216460/148558062-cb09898e-52ac-46b0-a7d1-6763d4b8e60c.png">

**Comment by timonmerk** [comment]
Also, this command `pip install -r requirements.txt --editable . ` should by automatically handled by the `conda env create --file=env.yml`. 
In the .yml file the the requirements.txt is being installed

**Comment by richardkoehler** [comment]
Okay, I think I have a clearer picture now!
So my thoughts:

- I think the flag "--user" actually simply doesn't exist for conda env create: https://docs.conda.io/projects/conda/en/latest/commands/create.html
- Okay, I see the problem now that occured with the __init__.py file. The reason that it wasn't working for me was because I tried the following:
> import py_neuromodulation as nm
> nm_bids = nm.nm_BidsStream.BidsStream()
- This is probably not possible if the __init__.py file is empty. However, this might be a problem, as the way that many users import packages is something like 
>import numpy as np
>import pandas as pd
- We might want to think about making this possible by importing the modules in the __init__.py file:
> from . import nm_BidsStream
> from . import ...
- The problem with the current conda environment setup is the command in the env.yml file:
>   - pip:
>       - -r requirements.txt
- Conda is handing over the installation responsibility to pip, which does not check if the packages are compatible on your given platform. This made automatic environment setup for myself (and likely many other users impossible). I made a suggestion on how to handle to modify the env.yml to fix this issue in the branch I posted. Conda now handles the installation of all packages, except for pybids, which is not available via conda, only via pip. The requirements.txt for installation of py_neuromodulation remains untouched.
